### ISSUE TITLE
batch size with multiple gpus

### ISSUE THREAD

**Issue by ylmeng** [pytorch/fairseq#2105]
A quick question.
If batch size is 100 for fairseq-train, and 2 GPUs are used. 
(1) Does that mean each GPU gets a batch of 100? (It looks so.)
(2) Does the model update parameters after processing 100 instances or 200? (It seems to be 200).

Thanks.

**Comment by myleott** [comment]
(1) Yes
(2) 200. Each GPU will get a batch of 100 instances (200 instances total for 2 GPUs), do a forward/backward pass, sum the gradients, then update the parameters

**Comment by longkhanh-fam** [comment]
Hi @myleott , i encounter the problem that, if im traning no problem with 1 GPU, however, if i increase the GPU to more than 1, the sample list is empty. Should I increase the batch size? Thank you for your help
